<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Olivia Angelin-Bonnet" />

<meta name="date" content="2019-05-20" />

<title>sismonr tutorial</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">sismonr tutorial</h1>
<h4 class="author">Olivia Angelin-Bonnet</h4>
<h4 class="date">2019-05-20</h4>



<p>This document is a tutorial demonstrating how to use the <code>sismonr</code> package.</p>
<div id="getting-started" class="section level1">
<h1>Getting started</h1>
<div id="installing-julia" class="section level2">
<h2>Installing Julia</h2>
<p>To install Julia, go to <a href="https://julialang.org/downloads/" class="uri">https://julialang.org/downloads/</a> and follow the instructions. In brief:</p>
<ul>
<li><strong>Windows</strong>: download the <code>.exe</code> file and run it to install Julia.</li>
<li><strong>MacOS</strong>: download the <code>.dmg</code> file, which contains Julia.app. Copy the later to your hard-drive or run from the disk image.</li>
<li><strong>Linux</strong>: download the <code>.tar.gz</code> file, and extract it to a folder of your choice.</li>
</ul>
</div>
<div id="adding-julia-to-the-path-environment-variable" class="section level2">
<h2>Adding Julia to the <code>PATH</code> environment variable</h2>
<p>For <code>sismonr</code> to be able to use Julia, the Julia executable must be in the <code>PATH</code> environment variable. You can add it as follow:</p>
<ul>
<li><strong>Windows</strong>: open the Control Panel and go to <code>System &gt; Advanced system settings &gt; Environment variables</code>. Select the <code>PATH</code> variable, click on <code>Edit &gt; New</code> and copy-paste the path <code>[path_to_julia_folder]/bin</code>, replacing the <code>[path_to_julia_folder]</code> with the path to the Julia directory.</li>
<li><strong>MacOS and Linux</strong>: in the terminal, type <code>sudo ln -s path_to_julia_folder/bin/julia /usr/local/bin/julia</code>, replacing the <code>[path_to_julia_folder]</code> with the path to the Julia directory.</li>
</ul>
<p>After this step it is necessary to restart your computer.</p>
</div>
<div id="installing-sismonr" class="section level2">
<h2>Installing sismonr</h2>
<p>From R, you can install the latest version of <code>sismonr</code> available on Github at <a href="https://github.com/oliviaAB/sismonr" class="uri">https://github.com/oliviaAB/sismonr</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span>(<span class="op">!</span><span class="kw">require</span>(<span class="st">&quot;devtools&quot;</span>)) <span class="kw">install.packages</span>(<span class="st">&quot;devtools&quot;</span>); <span class="kw">library</span>(devtools)
<span class="kw">install_github</span>(<span class="st">&quot;oliviaAB/sismonr&quot;</span>)</code></pre></div>
<p>If Julia is not installed on your computer or the Julia executable is not in your <code>PATH</code> environment variable, you won’t be able to install <code>sismonr</code>.</p>
</div>
<div id="sismonr-resources" class="section level2">
<h2>sismonr resources</h2>
<p>An online documentation is available at <a href="https://oliviaab.github.io/sismonr/" class="uri">https://oliviaab.github.io/sismonr/</a>.</p>
</div>
</div>
<div id="sismonr" class="section level1">
<h1>sismonr</h1>
<p>In order to use <code>sismonr</code> we need to load the package. You will notice that every time you load the package, <code>sismonr</code> checks that the necessary Julia modules are installed. If not, it automatically installs them for you.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(sismonr)</code></pre></div>
<div id="creating-an-in-silico-system" class="section level2">
<h2>1. Creating an <em>in silico</em> system</h2>
<p>The first step to simulate gene expression data is to generate an <em>in silico</em> system. The <em>in silico</em> system contains the list of genes whose expression we want to simulate along with the Gene Regulatory Network (GRN) representing the regulatory interactions among the genes. We can generate a random <em>in silico</em> system with:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myinsilicosystem =<span class="st"> </span><span class="kw">createInSilicoSystem</span>(<span class="dt">G =</span> <span class="dv">10</span>, <span class="dt">PC.p =</span> <span class="fl">0.7</span>)</code></pre></div>
<p>The different parameters passed to the function allow the user to control different aspects of the generated system. For example, <code>G</code> defines the number of genes present in the system. <code>PC.p</code> gives the probability for each gene to be protein-coding (so setting <code>PC.p = 1</code> ensures that there are only protein-coding genes in the system). A list of all parameters can be seen in the <a href="https://oliviaab.github.io/sismonr/#parameters-for-in-silico-system-generation">online documentation</a> of <code>sismonr</code> or by typing:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">?insilicosystemargs</code></pre></div>
<p>We’ll see the effect of some of these parameters in the next sections.</p>
<p>The function returns an object of class <code>insilicosystem</code>, which is a list containing the different attributes of the system:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">class</span>(myinsilicosystem)
<span class="kw">names</span>(myinsilicosystem)</code></pre></div>
<p>The next sections allow you to explore the <em>in silico</em> system we created.</p>
<div id="the-genes" class="section level3">
<h3>The genes</h3>
<p>The list of genes in the <em>in silico</em> system are stored in the <code>genes</code> element:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myinsilicosystem<span class="op">$</span>genes</code></pre></div>
<p>You can try generating new system by changing some parameters affecting the genes:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## system with only protein-coding genes, all regulators of transcription (PC.TC.p),
## and all regulations are activations (positive regulation - TC.pos.p)
myinsilicosystem2 =<span class="st"> </span><span class="kw">createInSilicoSystem</span>(<span class="dt">G =</span> <span class="dv">15</span>, <span class="dt">PC.p =</span> <span class="dv">1</span>, <span class="dt">PC.TC.p =</span> <span class="dv">1</span>, <span class="dt">TC.pos.p =</span> <span class="dv">1</span>) 
myinsilicosystem2<span class="op">$</span>genes

## Changing the function used to sample transcription rates for the genes
myinsilicosystem3 =<span class="st"> </span><span class="kw">createInSilicoSystem</span>(<span class="dt">G =</span> <span class="dv">10</span>,
                                         <span class="dt">basal_transcription_rate_samplingfct =</span> <span class="cf">function</span>(x){<span class="kw">runif</span>(x, <span class="fl">0.1</span>, <span class="fl">0.8</span>)})
myinsilicosystem3<span class="op">$</span>genes</code></pre></div>
</div>
<div id="the-grn" class="section level3">
<h3>The GRN</h3>
<p>The list of regulatory interactions in the system is stored in the <code>edg</code> element:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myinsilicosystem<span class="op">$</span>edg</code></pre></div>
</div>
<div id="the-regulatory-complexes" class="section level3">
<h3>The regulatory complexes</h3>
<p>The list of regulatory complexes in the system is stored in the <code>complexes</code> element. Their kinetic parameters are stored in the <code>complexeskinetics</code> element:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myinsilicosystem<span class="op">$</span>complexes
myinsilicosystem<span class="op">$</span>complexeskinetics</code></pre></div>
<p>It is likely to be empty if the number of genes is small, as regulatory complexes are generated from regulators targeting a common gene in the GRN. You can try increasing the number of genes in the system so see some complexes.</p>
<p>The <code>complexesTargetReaction</code> object gives the biological function of each complex, with the same code than for genes (e.g. “TC” means regulator of transcription).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myinsilicosystem<span class="op">$</span>complexesTargetReaction</code></pre></div>
</div>
</div>
<div id="plotting-the-grn" class="section level2">
<h2>Plotting the GRN</h2>
<p>You can visualise the GRN in your system with:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotGRN</span>(myinsilicosystem)</code></pre></div>
<p>By default all the types of regulation are drawn. You can also plot only a specific type of regulation:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotGRN</span>(myinsilicosystem, <span class="dt">edgeType =</span> <span class="st">&quot;TC&quot;</span>)</code></pre></div>
<div id="kinetic-properties-of-the-regulatory-interactions" class="section level3">
<h3>Kinetic properties of the regulatory interactions</h3>
<p>The <code>edg</code> element contains all the regulations occuring in the system. However each type of regulation is characterised by a distinct set of kinetic properties (e.g. to model a transcription regulation we need to know the binding and unbinding rates of the regulator to the target binding site, and the fold-change induced in the target transcription rate when the regulator is bound). This information is stored in a set of data-frames, one for each type of regulation. These data-frames are found in the <code>mosystem</code> element of the system:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(myinsilicosystem<span class="op">$</span>mosystem)
myinsilicosystem<span class="op">$</span>mosystem<span class="op">$</span>TCRN_edg</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">myinsilicosystem<span class="op">$</span>mosystem<span class="op">$</span>TLRN_edg
myinsilicosystem<span class="op">$</span>mosystem<span class="op">$</span>RDRN_edg
myinsilicosystem<span class="op">$</span>mosystem<span class="op">$</span>PDRN_edg
myinsilicosystem<span class="op">$</span>mosystem<span class="op">$</span>PTMRN_edg</code></pre></div>
</div>
</div>
<div id="creating-an-in-silico-population" class="section level2">
<h2>2. Creating an <em>in silico</em> population</h2>
<p>The second step to simulate gene expression data is to generate an <em>in silico</em> population. In a population, different variants of each gene are segregating (by default with equal frequency). Each gene variant is characterised by a set of QTL effect coefficient values that describe how genetic mutations affect the kinetic properties of the gene. Each <em>in silico</em> individual carries one or more copies of each gene (depending on the ploidy of the system), and these copies are sampled from the list of variants segregating in the population.</p>
<p>Here we create a population of diploid individuals, with 4 variants existing for each gene. We create 3 individuals:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypop =<span class="st"> </span><span class="kw">createInSilicoPopulation</span>(<span class="dv">3</span>, myinsilicosystem, <span class="dt">ngenevariants =</span> <span class="dv">4</span>, <span class="dt">ploidy =</span> <span class="dv">2</span>)</code></pre></div>
<p>The first parameter to be passed to <code>createInSilicoPopulation</code> is the number of individuals to be created, and the second is the <em>in silico</em> system we generated in the first step. Additional arguments can be given to the function, to control the properties of the individuals. A list of these arguments are available in the <a href="https://oliviaab.github.io/sismonr/#parameters-for-in-silico-individuals-generation">online documentation</a>, or with:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">?insilicoindividualargs</code></pre></div>
<p>The function returns an object of class <code>insilicopopulation</code>, which is a list containing the different attributes of the system:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">class</span>(mypop)
<span class="kw">names</span>(mypop)</code></pre></div>
<div id="the-gene-variants" class="section level3">
<h3>The gene variants</h3>
<p>The list of variants segregating in the population for each gene is stored in the <code>GeneVariants</code> element of the <em>in silico</em> population.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypop<span class="op">$</span>GenesVariants</code></pre></div>
<p>You can see the effect of changing the parameters passed to the <code>createInSilicoPopulation</code> function. You can even supply your own list of segregating variants and the frequency of these variants:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
mypop2 =<span class="st"> </span><span class="kw">createInSilicoPopulation</span>(<span class="dv">3</span>, myinsilicosystem, <span class="dt">ngenevariants =</span> <span class="dv">2</span>, <span class="dt">ploidy =</span> <span class="dv">2</span>)
mypop2<span class="op">$</span>GenesVariants

## Creating a smaller system with only 3 genes
mysystem =<span class="st"> </span><span class="kw">createInSilicoSystem</span>(<span class="dt">G =</span> <span class="dv">3</span>, <span class="dt">PC.p =</span> <span class="dv">1</span>)

## We will create only 1 variant of gene 1, 3 variants of gene 2 and
## 2 variants of gene 3
nbvariants =<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">2</span>)

qtlnames =<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;qtlTCrate&quot;</span>, <span class="st">&quot;qtlRDrate&quot;</span>,
             <span class="st">&quot;qtlTCregbind&quot;</span>, <span class="st">&quot;qtlRDregrate&quot;</span>,
             <span class="st">&quot;qtlactivity&quot;</span>, <span class="st">&quot;qtlTLrate&quot;</span>,
             <span class="st">&quot;qtlPDrate&quot;</span>, <span class="st">&quot;qtlTLregbind&quot;</span>,
             <span class="st">&quot;qtlPDregrate&quot;</span>, <span class="st">&quot;qtlPTMregrate&quot;</span>)

genvariants =<span class="st"> </span><span class="kw">lapply</span>(nbvariants, <span class="cf">function</span>(x){
  <span class="kw">matrix</span>(<span class="dv">1</span>, <span class="dt">nrow =</span> <span class="kw">length</span>(qtlnames), <span class="dt">ncol =</span> x,
         <span class="dt">dimnames =</span> <span class="kw">list</span>(qtlnames, <span class="dv">1</span><span class="op">:</span>x))
})
<span class="kw">names</span>(genvariants) =<span class="st"> </span>mysystem<span class="op">$</span>genes<span class="op">$</span>id

## the 2nd variant of gene 2 has a mutation reducing its transcription rate by 3
genvariants<span class="op">$</span><span class="st">`</span><span class="dt">2</span><span class="st">`</span>[<span class="st">&quot;qtlTCrate&quot;</span>, <span class="dv">2</span>] =<span class="st"> </span><span class="fl">0.33</span>
## and the 3rd variant has an increased translation rate
genvariants<span class="op">$</span><span class="st">`</span><span class="dt">2</span><span class="st">`</span>[<span class="st">&quot;qtlTLrate&quot;</span>, <span class="dv">2</span>] =<span class="st"> </span><span class="fl">1.5</span>

## The 2nd variant of gene 3 has a mutation decreasing the activity of
## its active product
genvariants<span class="op">$</span><span class="st">`</span><span class="dt">3</span><span class="st">`</span>[<span class="st">&quot;qtlactivity&quot;</span>, <span class="dv">2</span>] =<span class="st"> </span><span class="fl">0.7</span>

## Allelic frequency of each variant
genvariants.freq =<span class="st"> </span><span class="kw">list</span>(<span class="st">'1'</span> =<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>),
                        <span class="st">'2'</span> =<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.6</span>, <span class="fl">0.3</span>, <span class="fl">0.1</span>),
                        <span class="st">'3'</span> =<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.9</span>, <span class="fl">0.1</span>))

mypop3 =<span class="st"> </span><span class="kw">createInSilicoPopulation</span>(<span class="dv">10</span>, mysystem,
                                 <span class="dt">genvariants =</span> genvariants,
                                 <span class="dt">genvariants.freq =</span> genvariants.freq)</code></pre></div>
</div>
<div id="the-in-silico-individuals" class="section level3">
<h3>The <em>in silico</em> individuals</h3>
<p>The list of <em>in silico</em> individuals created is stored in the <code>individualsList</code> element:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(mypop<span class="op">$</span>individualsList)</code></pre></div>
<p>Each individual is characterised by the variants that it carries for each gene:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypop<span class="op">$</span>individualsList<span class="op">$</span>Ind1<span class="op">$</span>haplotype
mypop<span class="op">$</span>individualsList<span class="op">$</span>Ind2<span class="op">$</span>haplotype
mypop<span class="op">$</span>individualsList<span class="op">$</span>Ind3<span class="op">$</span>haplotype</code></pre></div>
<p>As the individuals are diploid, they have two copies of each genes. The alleles are termed <code>GCN1</code> and <code>GCN2</code>. Each row represents a gene, and the values in the data-frame correspond to the variant numbers of the genes that the individual carries.</p>
<p>You can try to change the ploidy of the individuals:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mypop2 =<span class="st"> </span><span class="kw">createInSilicoPopulation</span>(<span class="dv">3</span>, myinsilicosystem, <span class="dt">ngenevariants =</span> <span class="dv">2</span>, <span class="dt">ploidy =</span> <span class="dv">4</span>)
mypop2<span class="op">$</span>individualsList<span class="op">$</span>Ind1<span class="op">$</span>haplotype
mypop2<span class="op">$</span>individualsList<span class="op">$</span>Ind2<span class="op">$</span>haplotype
mypop2<span class="op">$</span>individualsList<span class="op">$</span>Ind3<span class="op">$</span>haplotype</code></pre></div>
<p>You can visualise the QTL effect coefficients of each variant carried by the individuals:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotMutations</span>(mypop, myinsilicosystem, <span class="dt">nGenesPerRow =</span> <span class="dv">5</span>)</code></pre></div>
<p>The function <code>plotMutations</code> takes as arguments the <em>in silico</em> population and the <em>in silico</em> system, and plot the value (colour) of each QTL effect coefficient (x-axis) for each allele (y-axis) of each gene (columns) for each individual in the population (rows). As some QTL effect coefficients do not apply to noncoding genes (i.e. mutations that would affect the translation or protein life), they are greyed in the plot.</p>
<p>You can zoom on specific genes, individuals, alleles or sets of QTL effect coefficients:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotMutations</span>(mypop, myinsilicosystem,
              <span class="dt">qtlEffectCoeffs =</span> <span class="kw">c</span>(<span class="st">&quot;qtlTCrate&quot;</span>, <span class="st">&quot;qtlTLrate&quot;</span>, <span class="st">&quot;qtlRDrate&quot;</span>, <span class="st">&quot;qtlPDrate&quot;</span>),
              <span class="dt">inds =</span> <span class="kw">c</span>(<span class="st">&quot;Ind1&quot;</span>, <span class="st">&quot;Ind2&quot;</span>),
              <span class="dt">alleles =</span> <span class="st">&quot;GCN2&quot;</span>,
              <span class="dt">genes =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)</code></pre></div>
</div>
</div>
<div id="simulating-the-in-silico-system" class="section level2">
<h2>3. Simulating the <em>in silico</em> system</h2>
<p>To get the expression of the different genes for each <em>in silico</em> individual, we will perform a stochastic simulation:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sim =<span class="st"> </span><span class="kw">simulateInSilicoSystem</span>(myinsilicosystem, mypop, <span class="dt">simtime =</span> <span class="dv">1000</span>, <span class="dt">ntrials =</span> <span class="dv">5</span>)</code></pre></div>
<p>or with the parallelised version of the simulation function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sim =<span class="st"> </span><span class="kw">simulateParallelInSilicoSystem</span>(myinsilicosystem, mypop, <span class="dt">simtime =</span> <span class="dv">1000</span>, <span class="dt">ntrials =</span> <span class="dv">5</span>)</code></pre></div>
<p>The argument <code>simtime</code> allows you to control the simulation end time in seconds (here we simulate the expression of the genes for 1000 s). <code>ntrials</code> correspond to the number of repetitions of the simulation that will be computed for each individual (here set to 5).</p>
<p>The returned object contains the running time (in seconds) of each individual’s simulation.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sim<span class="op">$</span>runningtime</code></pre></div>
<p>The results of the simulation are stored as a data-frame in the <code>Simulation</code> element of the simulation output:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(sim<span class="op">$</span>Simulation)</code></pre></div>
<p>in which each row corresponding to one time-point of one run of a simulation for one <em>in silico</em> individual.</p>
<p>You can visualise the results of your simulations in the form of a time-series of gene expression with:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotSimulation</span>(sim<span class="op">$</span>Simulation)</code></pre></div>
<p>The abundance of the different species (separated by RNAs -bottom-, proteins -middle- and regulatory complexes -top-) are plotted over time. As the simulation has been repeated 5 times for each individual, the mean abundance of the molecules over the different repetitions or trials is plotted as a solid line, and the minimum and maximum values are represented by the coloured areas. By default the abundances are plotted on a log10 scale, but you can change that with the option <code>yLogScale = F</code> in the plotSimulation function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotSimulation</span>(sim<span class="op">$</span>Simulation, <span class="dt">yLogScale =</span> F)</code></pre></div>
<p>The legend is presented as a table that gives for each component (columns) the different forms in which it can be found. The component names that are numbers correspond to the gene IDs. We can find the different genes either as RNAs or proteins. Gene IDs with the prefix “PTM” correspond to modified forms of proteins. Components with a name starting with a “C” correspond to regulatory complexes.</p>
<p>You can focus the plot on certain individuals or time-period:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotSimulation</span>(sim<span class="op">$</span>Simulation, <span class="dt">inds =</span> <span class="kw">c</span>(<span class="st">&quot;Ind1&quot;</span>), <span class="dt">timeMin =</span> <span class="dv">200</span>, <span class="dt">timeMax =</span> <span class="dv">300</span>)</code></pre></div>
<p>You can also visualise the output of the simulation as a heatmap, with the same arguments as the <code>plotSimulation</code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotHeatMap</span>(sim<span class="op">$</span>Simulation)</code></pre></div>
<div id="transforming-the-simulation-results" class="section level3">
<h3>Transforming the simulation results</h3>
<p>In the results of the simulation, molecules originating from different alleles of a same gene are differentiated by the suffix <code>GCNi</code>. You can merge these molecules to obtain the abundance of the corresponding molecule regardless of its allele of origin with:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">simNoAllele =<span class="st"> </span><span class="kw">mergeAlleleAbundance</span>(sim<span class="op">$</span>Simulation)
<span class="kw">head</span>(simNoAllele)</code></pre></div>
<p>Similarly, you can merge non-modified and post-translationally modified versions of a same protein:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">simNoPTM =<span class="st"> </span><span class="kw">mergePTMAbundance</span>(simNoAllele)
<span class="kw">head</span>(simNoPTM)</code></pre></div>
<p>or the abundance of free and in-complex molecules:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">simNoComplex =<span class="st"> </span><span class="kw">mergeComplexesAbundance</span>(simNoAllele)
<span class="kw">head</span>(simNoComplex)</code></pre></div>
</div>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
